### ISSUE TITLE
Implement pytest suite and CI

### ISSUE THREAD

**Issue by Kartstig** [mass-matrix/AMPL#29]
## Background
It would be nice to have one script to run all test suites. This might take a long time to do, so it might be best to reserve this action for pull requests only (not every commit). It seems there are tests which will run from the `atomsci/ddm/test` folder but some may fail if we choose to run in CI. If we can utilize some test tagging system, then we can isolate tests which should always pass on anyone's machine. These tests should be run on CI either on-push or PR

## Continuous Integration
There is a placeholder in the `pytest.yml` file. It is currently blocked by the linter `ruff`. This is another pre-step to prevent the entire suite from running

## Notes
- Livermore-based tests may not work for those outside of this environment
- This docmentation should be updated if we choose to make any changes to this workflow: https://github.com/ATOMScience-org/AMPL?tab=readme-ov-file#advanced-testing
